libraryversions.toml

```diff
@@ -235,7 +235,7 @@ KRUTH = { group = "androidx.kruth", atomicGroupVersion = "versions.KRUTH" }
 LEANBACK = { group = "androidx.leanback" }
 LEGACY = { group = "androidx.legacy" }
 LIBYUV = { group = "libyuv", atomicGroupVersion = "versions.LIBYUV" }
-LIFECYCLE = { group = "androidx.lifecycle", atomicGroupVersion = "versions.LIFECYCLE" }
+LIFECYCLE = { group = "org.jetbrains.compose.lifecycle-internal", atomicGroupVersion = "versions.COMPOSE" , overrideInclude = [ ":lifecycle:lifecycle-common", ":lifecycle:lifecycle-runtime", ":lifecycle:lifecycle-runtime-compose" ]}
```

Comment:
Our plan is to move non-Compose libraries out of Compose versions and `-internal` suffix. It worked temporarily for `annotation`/`collection`, but it seems won't work for `lifecycle`, even temporarily. Because the purpose of `lifecycle` is to be available as public API (@MatkovIvan, please correct).

Do we need this now as a published library?

If we don't need it urgently, I would suggest to not merge `mpp/build.gradle.kts` yet, and do this:
- decide if we truly want to go path of different package/version
- decide how it will look for user (which artifact, which docs, which changelog)
- decide the package (`org.jetbrains.lifecycle`, `org.jetbrains.androidx.lifecycle`)
- decide if we should align release cycle with Compose (it will be much simpler)
- introduce a new versioning for non-Compose libraries
- setup CI for this versioning

I don't know which versioning we should use, but I would start from these requirements:
- the package shouldn't start with `org.jetbrains.compose`
- the first 2 number should be equal to the androidx library (it is 2.8 for lifecycle)
- it should be built as part of main CI pipeline (with Compose)
- there should be an easy way to define a stable version (for Compose it is set by CI) to build dev/stable/custom builds

Reply:
> Because the purpose of lifecycle is to be available as public API

True

> it should be built as part of main CI pipeline (with Compose)

Disagree with this one

Reply:
Updated. Please review a new commit